### PR TITLE
Removes more dependencies on pegasus geography_us_states table.

### PR DESCRIPTION
**Pull request by ashercodeorg**
In doing so, changes behavior (nil is now returned rather than the abbreviation for unknown abbreviations) and renames get_state_name_from_abbr to get_us_state_from_abbr and so on.

This is part of a series of PRs (see also #7604) to eliminate the DB table, the gsheet, and the CSV.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
FYI @tanyaparker.

**Comment by ashercodeorg** [comment]
aoby and tanyaparker: AFAICT, there is no harm in changing the behavior of get_us_state_from_abbr for invalid abbreviations. Do either of you have enough context to confirm this?

**Comment by aoby** [comment]
What was the old behavior for invalid abbreviations? 

Note the DB table is still used in some places (i.e. [here](https://github.com/code-dot-org/code-dot-org/blob/c7b06dd613657ac2f7b428a7f3b77a29562f1e3e/pegasus/sites.v3/code.org/public/district-partner.haml#L29-L29)). Would it be worth converting these all to use the enum and then removing the table entirely?

Otherwise lgtm.

**Comment by ashercodeorg** [comment]
That is the intent, I'll chip away at the direct DB dependencies in another PR. Thanks for pointing them out, I had noticed them and forgotten them.

The old behavior (see the deleted code) was to return the abbreviation if the abbreviation wasn't valid. It isn't apparently (looking at the HAML) whether we were making use of that behavior. If we were, I can (for example) add an optional default_return = nil parameter to the function, allowing callers to specify the  abbr as the default return if not found.

**Comment by aoby** [comment]
OIC, `return code unless state`. Any idea what the reason would be for persisting the invalid abbreviation? Maybe that was meant for an earlier time when not all state abbreviations were in the db? I agree with you it's cleaner to stick with the enum and don't accept invalid ones. @tanyaparker any insight?

File: lib/state_abbr.rb [hunk 1]
```diff
@@ -5,19 +5,19 @@ def get_uppercase_symbol(abbr)
...
 end
 
 # Returns the state name associated with the state (including Washington DC)
 # abbreviation abbr.
 # abbr: A two character (uppercase or lowercase) symbol or string.
-def get_state_name_with_dc_from_abbr(abbr)
+def get_us_state_with_dc_from_abbr(abbr)
   abbr = get_uppercase_symbol(abbr)
   if abbr == :DC
     return 'Washington DC'
```
Comment (joshlory) [inline lib/state_abbr.rb]: Can we add `DC: 'Washington DC'` to `STATE_ABBR_HASH` to get rid of this special case (and this method)?

Comment (joshlory) [inline lib/state_abbr.rb]: Never mind, I see in some cases we don't want DC.

**Comment by tanyaparker** [comment]
Looks like it's just used for the title of the page, and if it's nil it will just show the Country name. New behavior looks good to me.

**Pull request merged**